pkg/odo/cli/component/push.go

```diff
@@ -258,16 +260,22 @@ func (po *PushOptions) Validate() (err error) {
 
 // Run has the logic to perform the required actions as part of command
 func (po *PushOptions) Run(cmd *cobra.Command) (err error) {
-	scontext.SetClusterType(cmd.Context(), po.Client)
+	if segment.IsTelemetryEnabled(nil) {
```

Comment:
There are multiple calls to `segment.IsTelemetryEnabled(nil)` in this function. Can we store the return value of `segment.IsTelemetryEnabled(nil)` into some variable so that it can be reused?

Reply:
I was thinking the same, perhaps adding a property to cobra command or `Runnable` perhaps?

Reply:
There, done. Amended the multiple calls to `IsTelemetryEnabled` by adding its value to cmd.Context.